nifi-nar-bundles/nifi-stateless-processor-bundle/nifi-stateless-processor/src/main/java/org/apache/nifi/processors/stateless/ExecuteStateless.java

```diff
@@ -0,0 +1,1202 @@
+            @Override
+            public VersionedFlowSnapshot getFlowSnapshot() {
+                return flowSnapshot;
+            }
+
+            @Override
+            public String getFlowName() {
+                return flowSnapshot.getFlowContents().getName();
+            }
+
+            @Override
+            public Set<String> getFailurePortNames() {
+                return failurePortNames;
+            }
+
+            @Override
+            public List<ParameterContextDefinition> getParameterContexts() {
+                return null;
+            }
+
+            @Override
+            public List<ReportingTaskDefinition> getReportingTaskDefinitions() {
+                return Collections.emptyList();
+            }
+
+            @Override
+            public List<ParameterValueProviderDefinition> getParameterValueProviderDefinitions() {
+                return Collections.singletonList(parameterValueProviderDefinition);
+            }
+
+            @Override
+            public TransactionThresholds getTransactionThresholds() {
+                return transactionThresholds;
+            }
+        };
+    }
+
+    private StatelessEngineConfiguration createEngineConfiguration(final ProcessContext context, final int contentRepoIndex) {
+        final File workingDirectory = new File(context.getProperty(WORKING_DIRECTORY).getValue());
+        final File narDirectory = new File(context.getProperty(LIB_DIRECTORY).getValue());
+        final ResourceReference krb5Reference = context.getProperty(KRB5_CONF).asResource();
+        final File krb5Conf = krb5Reference == null ? null : krb5Reference.asFile();
+        final SSLContextService sslContextService = context.getProperty(STATELESS_SSL_CONTEXT_SERVICE).asControllerService(SSLContextService.class);
+
+        final SslContextDefinition sslContextDefinition;
+        if (sslContextService == null) {
+            sslContextDefinition = null;
+        } else {
+            sslContextDefinition = new SslContextDefinition();
+            sslContextDefinition.setKeyPass(sslContextService.getKeyPassword());
+            sslContextDefinition.setKeystoreFile(sslContextService.getKeyStoreFile());
+            sslContextDefinition.setKeystorePass(sslContextService.getKeyStorePassword());
+            sslContextDefinition.setKeystoreType(sslContextService.getKeyStoreType());
+            sslContextDefinition.setTruststoreFile(sslContextService.getTrustStoreFile());
+            sslContextDefinition.setTruststorePass(sslContextService.getTrustStorePassword());
+            sslContextDefinition.setTruststoreType(sslContextService.getTrustStoreType());
+        }
+
+        final String contentStorageStrategy = context.getProperty(CONTENT_STORAGE_STRATEGY).getValue();
+        final File contentRepoDirectory;
+        if (CONTENT_STORAGE_DISK.getValue().equals(contentStorageStrategy)) {
+            final File contentRepoRootDirectory = new File(workingDirectory, "flowfile-content");
```

Comment:
Is there any potential issue with using the same `flowfile-content` name for multiple processor invocations?  What do you think about including `ExecuteStateless` in the file name to help maintain tracking?

Reply:
@exceptionfactory multiple invocations of the same processor should use the same directory. Invocations from different processors should not. This is why the next line here create a new File that is a child of 'flowfile-content' and whose name is the identifier of the Processor. This allows us to separate out content repos per processor - and even per thread, actually, as multiple concurrent threads cannot be writing to the same content repo. I could add in  'execute-stateless' to the filename.

Reply:
Thanks for the clarification, that makes sense!